File: javascript/problem052.js
function check(n){
  var indexArray = [];
  var str = n.toString();
  for(a=0; a<str.length; a++){
    indexArray.push(str[a]);
  }
  indexArray.sort();

  for(i=2; i<=6; i++){
    var tempArray = [];
    var tempValue = n * i;
    var tempStr = tempValue.toString();
    for(j=0; j<tempStr.length; j++){
      tempArray.push(tempStr[j])
    }
    tempArray.sort();
    if(indexArray.toString() != tempArray.toString()){
      return false;
    }
  }
  return true;
}

var index = 0;

while(true){
  index = index + 1;
  if(check(index)){
    break;
  }
}

var result = index;
console.log(result);

File: python/problem052.py
def check(n):
    index_array = []
    for a in str(n):
        index_array.append(a)
    index_array.sort()

    for i in range(2, 7):
        temp_array = []
        temp_value = n * i
        for j in str(temp_value):
            temp_array.append(j)
        temp_array.sort()
        if str(index_array) != str(temp_array):
            return False
    return True


index = 0

while True:
    index += 1
    if check(index) is True:
        break

result = index
print(result)
